fire/inspectutils.py

```diff
@@ -24,6 +24,20 @@
 import six
 
 
+class _FullArgSpec(object):
```

Comment:
I think you are right, my hesitation is that once it is a public non-builtin type it should be able to instantiate itself from a function, and other things people will reasonably expect of it.

I suppose this isn't really a reason not to do it, but it will be trickier to decide what it's job is once it is publically visible.